url054c08669f_jules2689_CSI4124_tgz-pJ8-SMThemeParkExperi3J8/src/SMThemeParkExperi3.java

```diff
@@ -0,0 +1,206 @@
+		boolean[] fixBoardingTime = new boolean[] { true, true, false, false }; // fix_boarding_time
+																				// options
+																				// in
+																				// 4
+																				// cases
+
+		// Lets get a set of uncorrelated seeds
+		Seeds[] sds = new Seeds[NUMRUNS];
+		RandomSeedGenerator rsg = new RandomSeedGenerator();
+		for (int i = 0; i < NUMRUNS; i++) {
+			sds[i] = new Seeds(rsg);
+		}
+		// The other are Ck's and one more must be calculated to ensure the
+		// overall CF
+
+		double[] confLevels = computeLastCFLevel(cfLevels);
+
+		// Train Config
+		double[][] valuesCasesForTrains = new double[4][NUMRUNS];
+
+		// Cars Config
+		double[][] valuesCasesForCars = new double[4][NUMRUNS];
+
+		// Costs Config
+		double[][] valuesCasesForCosts = new double[4][NUMRUNS];
+		double[] valuesDiff21ForCosts = new double[NUMRUNS];
+		double[] valuesDiff31ForCosts = new double[NUMRUNS];
+		double[] valuesDiff41ForCosts = new double[NUMRUNS];
+
+		System.out.println("Running "+ NUMRUNS +" seeds.");
+		System.out.println("Confidence Levels: "+ confLevels);
+		System.out.println("\nPlease note that a large number of seeds may take a few hours to run.");
+		
+		// Number of Seeds
+		for (int i = 0; i < NUMRUNS; i++) {
+			Seeds seed = sds[i];
+			// For Each Case Scenario Within Cars Within Trains
+
+			// case 1
+			// For Each Train Scenario Within Seeds
+			traincarloop: for (int numTrains = Constants.MIN_NUMBER_OF_TRAINS; numTrains <= Constants.MAX_NUMBER_OF_TRAINS; numTrains++) {
+				// For Each Car Scenario Within Train
+				for (int numCars = Constants.MIN_NUMBER_OF_CARS * numTrains; numCars <= Constants.MAX_NUMBER_OF_CARS
+						* numTrains; numCars++) {
+					park = new SMThemePark(startTime, endTime, numTrains,
+							numCars, boardingOptions[0], fixBoardingTime[0],
+							seed, false);
+					park.runSimulation();
+					if (park.projectGoalReached()) {
+						valuesCasesForCosts[0][i] = park.output.getCost();
+						valuesCasesForCars[0][i] = numCars;
+						valuesCasesForTrains[0][i] = numTrains;
+						break traincarloop;
+					}
+				}
+			}
+
+			// case 2
+			// For Each Train Scenario Within Seeds
+			traincarloop: for (int numTrains = Constants.MIN_NUMBER_OF_TRAINS; numTrains <= Constants.MAX_NUMBER_OF_TRAINS; numTrains++) {
+				// For Each Car Scenario Within Train
+				for (int numCars = Constants.MIN_NUMBER_OF_CARS * numTrains; numCars <= Constants.MAX_NUMBER_OF_CARS
+						* numTrains; numCars++) {
+					park = new SMThemePark(startTime, endTime, numTrains,
+							numCars, boardingOptions[1], fixBoardingTime[1],
+							seed, false);
+					park.runSimulation();
+					if (park.projectGoalReached()) {
+						valuesCasesForCosts[1][i] = park.output.getCost();
+						valuesCasesForCars[1][i] = numCars;
+						valuesCasesForTrains[1][i] = numTrains;
+
+						valuesDiff21ForCosts[i] = valuesCasesForCosts[1][i]
+								- valuesCasesForCosts[0][i];
+						break traincarloop;
+					}
+				}
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 84:102, 106:124, 128:146.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.